### PR TITLE
Refactored opentracing module

### PR DESCRIPTION
**Pull request by whiskeysierra**
## Motivation and Context

It's now possible to control whether headers are injected to downstream dependencies. This is in the light of a recent incident where an external party was failing because opentracing headers were automatically considered for some kind of request hash and some signature validation failed. Someone might still want to have a client-side span with all tags etc., but not propagate the span context.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: riptide-opentracing/src/main/java/org/zalando/riptide/opentracing/NoOpActivation.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+@API(status = EXPERIMENTAL)
+public final class NoOpActivation implements Activation {
+
+    @Override
+    public Optional<Scope> activate(final Tracer tracer, final Span span) {
+        return Optional.empty();
+    }
+
+    @Override
+    public void close(final Scope scope) {
```
Comment (whiskeysierra) [inline riptide-opentracing/src/main/java/org/zalando/riptide/opentracing/NoOpActivation.java]: Codacy Issue found: [Document empty method body](https://app.codacy.com/app/whiskeysierra/riptide/pullRequest?prid=4528919)

**Comment by fatroom** [comment]
👍

**Comment by lukasniemeier-zalando** [comment]
:+1:

**Comment by whiskeysierra** [comment]
> Is it worth to document the new injection flexibility in the `README`?

Done

**Comment by lukasniemeier-zalando** [comment]
:+1:

**Pull request merged**